homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -299,19 +402,20 @@ async def _async_update_sources(self) -> None:
         # Combine the source dicts
         self._sources = self._audio_sources | self._video_sources
 
-        # HASS won't necessarily be running the first time this method is run
-        if self.hass.is_running:
-            self.async_write_ha_state()
+        self._attr_source_list = list(self._sources.values())
+
+        self.async_write_ha_state()
 
     @callback
-    def _async_update_playback_metadata(self, data: PlaybackContentMetadata) -> None:
+    async def _async_update_playback_metadata(
```

Comment:
Is the name of the method now misleading since it no longer just updates the image metadata?

Reply:
This is to show the correct friendly name in the extra state attribute `beolink` -> `self`

Reply:
Most of these method names refer to which WebSocket event triggers them, but yeah this should probably be changed.
Also, do you know if the `@callback` decorator can be used for async functions? It looks like it is only used 6 other places in Home Assistant and its use has been questioned here https://github.com/home-assistant/core/pull/121209#discussion_r1694788658